examples/gritlm/gritlm.cpp

```diff
@@ -44,6 +44,7 @@ static std::vector<std::vector<float>> encode(llama_context * ctx, const std::ve
 
         // clear previous kv_cache values (irrelevant for embeddings)
         llama_kv_cache_clear(ctx);
+        llama_set_embeddings(ctx, true);
```

Comment:
I have a small question here: in the case when both `embeddings` and `causal_attn` are both enabled, will it still be correct?

Reply:
In general, it's possible to run with `embeddings=true` and `causal_attn=true`, as long as the underlying model supports causal attention. For the GritLM case, I just checked here, and it will run but give incorrect results since it expects the embeddings to be run non-causally.